src/deepsparse/open_pif_paf/pipelines.py

```diff
@@ -45,8 +81,50 @@ def output_schema(self) -> Type[OpenPifPafOutput]:
     def setup_onnx_file_path(self) -> str:
         return self.model_path
 
-    def process_inputs(self, inp):
-        return inp.images
+    def process_inputs(self, inputs):
+        images = []
+
+        for image_path in inputs.images:
+            with open(image_path, "rb") as f:
+                image = PIL.Image.open(f).convert("RGB")
+                # second and third arguments are optional:
+                #    - gt annotation
+                #    - image metadata
+                image, _, _ = self.pre_process_transformations(image, [], {})
+                # add batch dimension and convert to numpy
+                images.append(image.numpy())
+
+        return [numpy.stack(images)], {"original_images": copy.deepcopy(inputs.images)}
+
+    def process_engine_outputs(self, fields, **kwargs):
+        """
+        Fields (raw network outputs) are a list of numpy arrays of sizes:
+        (B,17,5,13,17) (CIF) and (B,19,8,13,17) (CAF).
+        The processor maps fields into the actual list of pose annotations
+        """
+        for idx, (cif, caf) in enumerate(zip(*fields)):
+            annotations = self.processor._mappable_annotations(
+                [torch.tensor(cif), torch.tensor(caf)], None, None
+            )
```

Comment:
Ahh I see, so this is why we'd need to depend on openpifpaf? You said this is a pretty complicated function too right?

Reply:
I will do a more in-depth write up on this, but yeah, self.processor is basically a composition of decoders, each decoder is a fairly complex structure.